ui/packages/catalog-ui-search/src/main/webapp/component/visualization/table/table-viz.view.js

```diff
@@ -88,11 +112,37 @@ module.exports = Marionette.LayoutView.extend({
     if (!options.selectionInterface) {
       throw 'Selection interface has not been provided'
     }
+    this.filteredAttributes = new filteredAttributesModel()
+
     this.listenTo(
       this.options.selectionInterface,
       'reset:activeSearchResults add:activeSearchResults',
       this.handleEmpty
     )
+    this.listenTo(
+      this.options.selectionInterface,
+      'change:activeSearchResultsAttributes',
+      this.filterActiveSearchResultsAttributes
+    )
+  },
+  filterActiveSearchResultsAttributes() {
+    const currentQuery = this.options.selectionInterface.getCurrentQuery()
+
+    const resultFormName = currentQuery ? currentQuery.get('detail-level') : ''
+    const selectedResultTemplate = ResultForm.getResultCollection().filteredList.filter(
+      form => form.id === resultFormName || form.value === resultFormName
+    )[0]
```

Comment:
Could this ever be length 0?

Reply:
yeah, but it looks like the zero'th index of an empty array is just undefined

Reply:
https://stackoverflow.com/a/31235674
 🤔 
I'll probably just change it to avoid possible confusion in the future